posts/2021-03-25-Rust-1.51.0.md

```diff
@@ -0,0 +1,179 @@
+
+This change now lets us have our own array struct that's generic over its type *and* its length. Let's look an example definition, and how it can be used.
+
+```rust
+struct Array<T, const LENGTH: usize> {
+    //          ^^^^^^^^^^^^^^^^^^^ Const generic definition.
+    list: [T; LENGTH]
+    //        ^^^^^^ We use it here.
+}
+
+// Array<u8, 32>
+// Becomes 
+// ```
+// struct Array_u8_32 {
+//     list: [u8; 32]
+// }
+// ```
+
+impl<T, const LENGTH: usize> Array<T, LENGTH> {
+    fn last(&self) -> Option<&T> {
+        if LENGTH == 0 {
+        // ^^^^^^^^^^^ This is always `true` for `Array<T, 0>`.
+            None
+        } else {
+            Some(self.list[LENGTH - 1])
+        }
+    }
+}
+```
+
+Const generics adds an important new tool for library designer's in creating new powerful compile-time safe APIs. If you'd like to learn more about const generics you can also check out the ["Const Generics MVP Hits Beta"][const-generics-blog] blog post for more information about the feature and its current restrictions. We can't wait to see what new libraries and APIs you create!
+
+[const-generics-blog]: https://blog.rust-lang.org/2021/02/26/const-generics-mvp-beta.html
+
+### Cargo's New Feature Resolver
+Dependency management is a hard problem, and one of the hardest parts of it is just picking what *version* of a dependency to use when it's depended on by two different packages, this doesn't just include its version number, but also what features are or aren't enabled for the package. Cargo's default behaviour is to merge features for a single package. Let's say you if you had a dependency called `foo` with features A and B, which was being used by packages `bar` and `baz`, but `bar` depends on `foo+A` and `baz` depends on `foo+B`, cargo will merge both of those features and compile `foo` as `foo+AB`. This has a benefit that you only have to compile `foo` once, and then it can reused for both `foo` and `bar`.
+
+However, this also comes with a downside, what if `A` and `B` are incompatible? What if your own code is compatible with `foo` but not `foo+A`? A common example of this in the ecosystem is the optional `std` feature included in many `#![no_std]` crates, that allows crates to provide added functionality when `std` is available. Now imagine you want to use the `#![no_std]` version of `foo` in your `#![no_std]` binary, and use the `foo` at build time in your `build.rs`, if your build time dependency depends on `foo+std`, your binary now also depends on `foo+std`, which means it will no longer compile because `std` is not available for your target platform. 
+
+This has been a long standing issue in cargo, and with this release there's a new `resolver` option in your `Cargo.toml`, where you can set `resolver="2"` to tell cargo to try a new approach to resolving features, you can check out [RFC 2957] a detailed description of the behaviour, but it can summarised as follows.
```

Comment:
How to enable the feature is a bit hidden in the middle of the sentence, while for other things that need to be enabled (like debuginfo on macOS) we provide a code snippet at the end of the section. Maybe we could do the same here?